### PR TITLE
Update themePropType

### PR DESCRIPTION
**Pull request by hris27**
## Purpose

Fix console warning about `typography.subHeadline` propType.

## Approach and changes

- Fixed `typography.subHeadline` propType.
- Updated some missing keys in other propTypes.

## Definition of done

* [ ] Development completed
* [ ] Reviewers assigned
* [ ] Unit and integration tests
* [ ] Meets minimum browser support
* [ ] Meets accessibility requirements

### CODE REVIEW

File: packages/design-tokens/utils/theme-prop-type.ts [hunk 1]
```diff
@@ -164,7 +164,7 @@ export const themePropType = PropTypes.shape({
       three: typePropType,
       four: typePropType,
     }).isRequired,
-    subHeadline: PropTypes.shape({ typePropType }).isRequired,
+    subHeadline: PropTypes.shape({ ...typePropType }).isRequired,
```
Comment (connor-baer) [inline packages/design-tokens/utils/theme-prop-type.ts:167]:
```suggestion
    subHeadline: PropTypes.shape(typePropType).isRequired,
```

File: .changeset/tidy-otters-promise.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@sumup/design-tokens': patch
+---
+
+Updated themePropType to match the current state of the Circuit UI theme.
```
Comment (connor-baer) [inline .changeset/tidy-otters-promise.md:5]:
```suggestion
Updated `themePropType` to match the current state of the Circuit UI theme.
```

**Review by connor-baer — COMMENTED** [review]
Thank you so much for making these updates! 🙌🏻

We've neglected the prop types a bit after the transition to TypeScript 😬

**Review by amelako — APPROVED** [review]
Thanks for cleaning it up 🌟

File: packages/design-tokens/utils/theme-prop-type.ts (hunk 1 above)
Comment (connor-baer) [reply packages/design-tokens/utils/theme-prop-type.ts]: Hm, apparently TypeScript is unhappy about this 🙄

Comment (connor-baer) [reply packages/design-tokens/utils/theme-prop-type.ts]: Feel free to revert.

Comment (hris27) [reply packages/design-tokens/utils/theme-prop-type.ts]: It should be just `subHeadline: typePropType` 😅

**Review by connor-baer — APPROVED** [review]

**Pull request merged**